Mlem/Models/Trackers/Feed/TrackerItem.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  TrackerSortableNew.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2023-10-15.
+//
+import Foundation
+
+protocol TrackerSortable {
+    func shouldSortBefore(_ other: Self?) -> Bool
+}
+
+protocol TrackerItem: Equatable {
```

Comment:
How would this system work with a model that needs to be used in multiple different parent trackers? For example, `PostModel` would be used in both the user profile and search trackers.

Something like this could work perhaps?

```swift
import Foundation


protocol SortType: RawRepresentable where RawValue: StringProtocol { }
                        
protocol ParentTrackerItem {
    associatedtype Sort: SortType
}

enum ProfileSortType: String, SortType {
    case published
}

enum SearchSortType: String, SortType {
    case searchScore
}

// This is the equivalent of AnyInboxItem
enum AnyProfileItem: ParentTrackerItem {
    // ...
    typealias Sort = ProfileSortType
}

class PostModel {
    // ...
    func sortValue(sortType: any SortType) -> Int {
        switch sortType {
        case ProfileSortType.published:
            return self.published
        case SearchSortType.searchScore:
            return self.score
        default:
            fatalError("postModel does not support this sort type!")
        }
    }
}
```

Reply:
Oops, that protocol was an artifact of a previous iteration that snuck through.

Sorting is the biggest bump in the carpet for the whole system--for any given parent/tracker pairing, the associated `Item` of the child needs to implement the same sorting as all of the other children of the parent. I tried a bunch of ways to make this work nicely, but that's a degree of nuance that the generics system just can't quite grasp. 

My solution is more or less what you've suggested, but with only the single `TrackerSortType`, which corresponds to `TrackerSortVal`, which in turn defines a custom comparison operator that performs runtime type safety checks in lieu of compiler-enforced.

In `TrackerItem` conformance extensions, the models implement that sort value switch. I went with that instead of a more granular per-feed sort type system because there's a fair amount of overlap--profile and feed, for example, should both be sortable on all of the common sorting operations between posts and comments, so it seems a bit redundant to define `PostSortType.new` and `CommentSortType.new`

I've gone through and made this cleaner--`TrackerItem` conformance is now its own extension separate from `InboxItem` conformance for the three models, and I've cleaned out the vestigial sorting logic.

Reply:
> I went with that instead of a more granular per-feed sort type system because there's a fair amount of overlap--profile and feed, for example, should both be sortable on all of the common sorting operations between posts and comments, so it seems a bit redundant to define PostSortType.new and CommentSortType.new

Ah yes, that makes sense. Looks good to me